### PR TITLE
Supports more authentication approaches on Azure in Storage Initializer

### LINKED ISSUE #2013: Support more authentication approaches on Azure in Storage Initializer
/kind feature

**Describe the solution you'd like**
Currently only Azure Service Principal login is supported in Storage Initializer.
Customer is not able to use Managed Identity to download models from Azure Blobs.

**Anything else you would like to add:**
Azure Identity Python SDK supports variety of authentication approaches.
We may simply leverages the feature provided by the SDK.

### PR DESCRIPTION
**Pull request by laozc**
**What this PR does / why we need it**:
Supports more authentication approaches on Azure in Storage Initializer

**Which issue(s) this PR fixes**:
Fixes #2013

**Special notes for your reviewer**:

**Release note**:
```release-note
Supports more authentication approaches on Azure in Storage Initializer
```

### CODE REVIEW

File: docs/samples/storage/azure/README.md [hunk 1]
```diff
@@ -51,4 +54,28 @@ kubectl apply -f azcreds.yaml
...
+
+You may assign Managed Identity to `InferenceService` resource using Pod Identity.
+
+* Set up Pod Identity follow the steps [here](https://azure.github.io/aad-pod-identity/docs/demo/standard_walkthrough/)
+* Add the `aadpodidbinding` label to your service 
+
+```yaml
+---
+apiVersion: serving.knative.dev/v1
+kind: Service
```
Comment (yuzisun) [inline docs/samples/storage/azure/README.md]: Should use the InferenceService CR here?

File: docs/samples/storage/azure/README.md [hunk 2]
```diff
@@ -51,4 +54,28 @@ kubectl apply -f azcreds.yaml
...
+  name: kserve-simple-string
+  labels:
+    aadpodidbinding: piselector
+spec:
+  template:
+    metadata:
+  predictor:
+    serviceAccountName: sa
+    tensorflow:
+      storageUri: "https://kserve.blob.core.windows.net/triton/simple_string/"
```
Comment (yuzisun) [inline docs/samples/storage/azure/README.md:80]: Does this url exist?

File: docs/samples/storage/azure/README.md [hunk 3]
```diff
@@ -23,16 +24,18 @@ metadata:
...
 type: Opaque
 data:
-  AZ_CLIENT_ID: xxxxx
-  AZ_CLIENT_SECRET: xxxxx
-  AZ_SUBSCRIPTION_ID: xxxxx
-  AZ_TENANT_ID: xxxxx
+  AZURE_CLIENT_ID: xxxxx
+  AZURE_CLIENT_SECRET: xxxxx
+  AZURE_SUBSCRIPTION_ID: xxxxx
+  AZURE_TENANT_ID: xxxxx
```
Comment (yuzisun) [inline docs/samples/storage/azure/README.md:30]: Are these standard names for Azure ?

File: pkg/credentials/azure/azure_secret.go [hunk 4]
```diff
@@ -21,58 +21,49 @@ import (
...
-			Name: AzureClientSecret,
+	var envs []v1.EnvVar
+	for _, k := range AzureEnvKeys {
+		dataKey := k
+		legacyDataKey := legacyAzureEnvKeyMappings[k]
+		if _, ok := secret.Data[legacyDataKey]; ok {
+			dataKey = legacyDataKey
+		}
+		envs = append(envs, v1.EnvVar{
+			Name: k,
```
Comment (yuzisun) [inline pkg/credentials/azure/azure_secret.go:58]: So the environment variables are no longer needed?

File: docs/samples/storage/azure/README.md (hunk 2 above)
Comment (laozc) [reply docs/samples/storage/azure/README.md:80]: It doesn't exist.
I don't think the previous one https://kfserving.blob.core.windows.net/triton/simple_string/ work as well but the kfserving.blob.core.windows.net exists.
Do we need to change the naming to kserve.blob.core.windows.net or stay in kfserving.blob.core.windows.net?

**Comment by yuzisun** [comment]
thanks @laozc for the awesome contributions!

/lgtm
/approve

**Pull request merged**